features/netsocket/nsapi_dns.cpp

```diff
@@ -221,30 +410,55 @@ static nsapi_size_or_error_t nsapi_dns_query_multiple(NetworkStack *stack, const
 
     nsapi_size_or_error_t result = NSAPI_ERROR_DNS_FAILURE;
 
+    bool retry = false;
+
+    int index = 0;
+
     // check against each dns server
-    for (unsigned i = 0; i < DNS_SERVERS_SIZE; i++) {
+    while (true) {
+        SocketAddress dns_addr;
+        err = nsapi_dns_get_server_addr(stack, &index, &dns_addr);
+        if (err != NSAPI_ERROR_OK) {
+            break;
+        }
+
         // send the question
-        uint8_t *question = packet;
-        dns_append_question(&question, host, version);
+        int len = dns_append_question(packet, 1, host, version);
 
-        err = socket.sendto(SocketAddress(dns_servers[i], 53), packet, question - packet);
+        err = socket.sendto(dns_addr, packet, len);
         // send may fail for various reasons, including wrong address type - move on
         if (err < 0) {
+            // goes to next dns server
+            retry = false;
+            index++;
             continue;
         }
 
         // recv the response
         err = socket.recvfrom(NULL, packet, DNS_BUFFER_SIZE);
         if (err == NSAPI_ERROR_WOULD_BLOCK) {
+            if (!retry) {
+                // retries once
```

Comment:
I've been experimenting with wifi on a bad connection and I often need to do multiple DNS lookups for it to go through. Would it be possible to make the retry number configurable and not fixed to one, please?

Reply:
I'll add json configuration options about retries.